src/test/java/ch/jalu/configme/configurationdata/CommentsConfigurationTest.java

```diff
@@ -11,12 +11,31 @@
 import static org.hamcrest.Matchers.contains;
 import static org.hamcrest.Matchers.containsInAnyOrder;
 import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 /**
  * Test for {@link CommentsConfiguration}.
  */
 class CommentsConfigurationTest {
 
+    @Test
+    void shouldThrowForExistingPath() {
+        // given
+        final String path = "config.me"; 
+        CommentsConfiguration conf = new CommentsConfiguration();
+        conf.setComment(path, "Old", "Comments", "1", "2", "3");
+        
+        // when
+        IllegalStateException ex = assertThrows(IllegalStateException.class, () -> conf.setComment(path, "New Comment"));
+
+        // then
+        assertThat(ex.getMessage(), equalTo("Comment lines already exists for the path " + path));
+        assertEquals(conf.getAllComments().size(), 1);
+        assertEquals(conf.getAllComments().get(path).size(), 1);
+        assertThat(conf.getAllComments().get(path), contains("New Comment"));
+    }
+
     @Test
     void shouldOverrideExistingComment() {
```

Comment:
I think this test method can be deleted in favor of the new method you've introduced

Reply:
This test method `shouldOverrideExistingComment` not only checks for comment lines overriding, but also checks that the other path is left unchanged (L53). Kindly suggest if it still seems like a duplicate test case and should be removed.

Reply:
That's true, that means the new test is a subset of the old test since it tests the same things, minus checking that other comments are unchanged. I think it's fine to keep both for now since there's no need for another iteration. Thanks for your work!